SemanticUI/Modules/Dropdown.elm

```diff
@@ -0,0 +1,456 @@
+                :: children
+            )
+        )
+        ctx.drawerIsOpen
+        ctx.dropdownCfg
+
+
+{-| Create an element for the toggle area that will toggle the drawer on toggle event.
+
+It takes the following:
+
+  - `Config`
+  - `State`
+  - Node type as a render function
+  - Attributes of node
+  - Children of node
+
+-}
+toggle : Config r msg -> State -> Render msg -> List (Attribute msg) -> List (Html msg) -> Html msg
+toggle cfg state render attrs children =
+    let
+        ctx =
+            toContext cfg state
+    in
+    if cfg.readOnly then
+        render attrs children
+
+    else
+        Dropdown.toggle render attrs children ctx.drawerIsOpen ctx.dropdownCfg
+
+
+{-| Create the drawer element which is displayed when the dropdown is expanded/toggled.
+
+It takes the following:
+
+  - `Config`
+  - `State`
+  - Node type as a render function e.g. `Html.div`
+  - Attributes of node
+  - Children of node (some of which atleast created with `item`)
+
+It adds the "menu" class to the node.
+
+-}
+drawer : Config r msg -> State -> Render msg -> List (Attribute msg) -> List (Html msg) -> Html msg
+drawer cfg state render attrs children =
+    let
+        ctx =
+            toContext cfg state
+    in
+    if cfg.readOnly then
+        render
+            (classList
+                [ ( "menu", True )
+                ]
+                :: attrs
+            )
+            children
+
+    else
+        Dropdown.drawer render
+            ((classList
+                [ ( "menu", True )
+                , ( "active", True )
+                , ( "visible", ctx.drawerIsVisible )
+                , ( "hiden", not ctx.drawerIsVisible )
+                , ( "animating", ctx.drawerIsTransitioning )
+                , ( "transition", True )
+                , ( "slide", ctx.drawerIsTransitioning )
+                , ( "down", ctx.drawerIsTransitioning )
+                , ( "in", state == Opening )
+                , ( "out", state == Closing )
+                ]
+                :: (if ctx.drawerIsTransitioning then
+                        [ on "animationend" (Decode.succeed (cfg.onToggle <| ctx.drawerToggledFinal)) ]
+
+                    else
+                        []
+                   )
+             )
+                ++ attrs
+            )
+            children
+            ctx.drawerIsOpen
+            ctx.dropdownCfg
+
+
+{-| Create an item that goes in the drawer.
+
+It takes the following:
+
+  - `Config`
+  - `State`
+  - Node type as a render function e.g. `Html.div`
+  - Attributes of node
+  - Children of node
+
+It adds the "item" class to the node.
+
+-}
+item : Render msg -> List (Attribute msg) -> List (Html msg) -> Html msg
+item render attrs children =
+    render (class "item" :: attrs) children
+
+
+type alias Context msg =
+    { drawerIsOpen : Bool
+    , drawerIsTransitioning : Bool
+    , drawerIsVisible : Bool
+    , drawerToggledFinal : State
+    , dropdownCfg : Dropdown.Config msg
+    }
```

Comment:
Not sure about the name `Context`, this seems like `State` to me. Should `State` be renamed to `DrawerState` perhaps?

Reply:
Ah right I see now that this is derived state. Still feels a little odd to me that it's mixed with config here... Feels like config should be static...